system/greenhouse-ccloud/templates/_helper.tpl

```diff
@@ -0,0 +1,31 @@
+{{/* Get related Thanos store endpoints */}}
+{{- define "thanos.getRelatedStoreEndpoints" -}}
+{{- $prefixes := index . 0 -}}
+{{- $root := index . 1 -}}
+{{- $stores := list }}
+{{- range $plugin := $root.Values.ingressPlugins -}}
+{{- range $prefix := $prefixes -}}
+{{- $thanosStore := printf "thanos-grpc.%s:443" (trimSuffix "." (trimPrefix "ingress." $plugin.recordName)) -}}
+{{- if contains $prefix $thanosStore -}}
+{{- $stores = append $stores $thanosStore -}}
```

Comment:
I'd like to ensure to actually have the format ( `ingress.XYZ.fqdn.`) in place to not get any garbage in which somehow contains the prefix.

```suggestion
{{- if and (hasPrefix "ingress" $plugin.recordName ) (hasSuffix "." $plugin.recordName ) }}
{{- $thanosStore := printf "thanos-grpc.%s:443" (trimSuffix "." (trimPrefix "ingress." $plugin.recordName)) -}}
{{- if contains $prefix $thanosStore -}}
{{- $stores = append $stores $thanosStore -}}
{{- end -}}
```

Reply:
I think the format is guaranteed anyway. But we'll take it